src/Interpreters/TableJoin.h

```diff
@@ -346,12 +347,14 @@ class TableJoin
     std::shared_ptr<const IKeyValueEntity> getStorageKeyValue() { return right_kv_storage; }
 
     /// proton : starts
-    void addLagBehindKeys(const ASTPtr & left_table_ast, const ASTPtr & right_table_ast);
+    void addTimestampKeys(const ASTPtr & left_table_ast, const ASTPtr & right_table_ast);
+    void setRequiredJoinAlignment(bool required_join_alignment_ = true) noexcept { required_join_alignment = required_join_alignment_; }
```

Comment:
JoinAlignment shall be deduced automatically, so it seems this flag can be removed.